app/src/main/java/io/homeassistant/companion/android/sensors/TrafficStatsManager.kt

```diff
@@ -0,0 +1,173 @@
+        val cm: ConnectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val networkInfo = cm.allNetworks
+        var networkCapabilities: NetworkCapabilities
+        for (item in networkInfo) {
+            networkCapabilities = cm.getNetworkCapabilities(item)
+            if (!hasCellular)
+                hasCellular = networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
+        }
+        return true
+    }
+    override fun requestSensorUpdate(
+        context: Context
+    ) {
+        updateMobileRxBytes(context)
+        updateMobileTxBytes(context)
+        updateTotalRxBytes(context)
+        updateTotalTxBytes(context)
+    }
+
+    private fun updateMobileRxBytes(context: Context) {
+
+        if (!isEnabled(context, rxBytesMobile.id))
+            return
+
+        var mobileRx = 0f
+        val icon = "mdi:radio-tower"
+
+        try {
+            mobileRx = TrafficStats.getMobileRxBytes().toFloat()
+        } catch (e: Exception) {
+            Log.e(TAG, "Error getting the mobile rx bytes", e)
+        }
+
+        mobileRx /= GB
+        onSensorUpdated(context,
+            rxBytesMobile,
+            mobileRx.toBigDecimal().setScale(3, RoundingMode.HALF_EVEN),
+            icon,
+            mapOf()
+        )
+    }
+
+    private fun updateMobileTxBytes(context: Context) {
+
+        if (!isEnabled(context, txBytesMobile.id))
+            return
+
+        var mobileTx = 0f
+        val icon = "mdi:radio-tower"
+
+        try {
+            mobileTx = TrafficStats.getMobileTxBytes().toFloat()
+        } catch (e: Exception) {
+            Log.e(TAG, "Error getting the mobile tx bytes", e)
+        }
+
+        mobileTx /= GB
+        onSensorUpdated(context,
+            txBytesMobile,
+            mobileTx.toBigDecimal().setScale(3, RoundingMode.HALF_EVEN),
+            icon,
+            mapOf()
+        )
+    }
+    private fun updateTotalRxBytes(context: Context) {
+
+        if (!isEnabled(context, rxBytesTotal.id))
+            return
+
+        var totalRx = 0f
+        val icon = "mdi:radio-tower"
+
+        try {
+            totalRx = TrafficStats.getTotalRxBytes().toFloat().absoluteValue
+        } catch (e: Exception) {
+            Log.e(TAG, "Error getting the total rx bytes", e)
+        }
+        totalRx /= GB
+
+        onSensorUpdated(context,
+            rxBytesTotal,
+            totalRx.toBigDecimal().setScale(3, RoundingMode.HALF_EVEN),
+            icon,
+            mapOf()
+        )
+    }
+
+    private fun updateTotalTxBytes(context: Context) {
+
+        if (!isEnabled(context, txBytesTotal.id))
+            return
+
+        var totalTx = 0f
+        val icon = "mdi:radio-tower"
+
+        try {
+            totalTx = TrafficStats.getTotalTxBytes().toFloat().absoluteValue
+        } catch (e: Exception) {
+            Log.e(TAG, "Error getting the total tx bytes", e)
+        }

```

Comment:
If we have an issue getting the value we shouldn't continue to update the sensor otherwise you'll get random `0 GB` sent.  Also do we need the `absoluteValue`?

Reply:
Same as above, my pixel was showing a negative value so just wanted to make it positive...it only occurs for total bytes lol